Examples/Examples/Util/PaymentStatusPoller.swift

```diff
@@ -0,0 +1,184 @@
+//
+//  PaymentStatusPoller.swift
+//  Examples
+//
+//  Created by Claude on 2025/12/04.
+//  Copyright © 2025 Airwallex. All rights reserved.
+//
+
+import Foundation
+import UIKit
+import Airwallex
+
+@MainActor
+protocol PaymentStatusPollerDelegate: AnyObject {
+    func paymentStatusPollerDidStartPolling(_ poller: PaymentStatusPoller)
+    func paymentStatusPoller(_ poller: PaymentStatusPoller, didUpdateStatus attempt: PaymentAttempt)
+    func paymentStatusPoller(_ poller: PaymentStatusPoller, didFailWithError error: Error)
+    func paymentStatusPoller(_ poller: PaymentStatusPoller, didTimeoutWithStatus attempt: PaymentAttempt)
+}
+
+@MainActor
+class PaymentStatusPoller {
```

Comment:
I'm wondering why we're not just expose a simple public method:
`func getPaymentAttempt() async throws -> PaymentAttempt`
So when VC calls it, it will start loading until it's final or throw error. I just feel we should design it simply enough for consumers to understand.

Reply:
If wants to help debug, we can just log the intermediate state to console inside this class but keep the interface simple.